haskell/test/SMT/Common/Run.hs

```diff
@@ -243,52 +233,52 @@ handleCallViaLigo (dao, gov, viewC) mc = do
   let fsE = case result of
         Just err -> Left err
         Nothing -> Right fs
 
   daoBalance <- getBalance dao
 
-  govStore <- getFullStorage @([FA2.TransferParams]) (unTAddress gov)
+  govStore <- getFullStorage @([FA2.TransferParams]) gov
   govBalance <- getBalance gov
 
-  viewStorage <- getFullStorage @([(MText, Maybe MText)]) (unTAddress viewC)
+  viewStorage <- getFullStorage @([(MText, Maybe MText)]) viewC
   pure (fsE, daoBalance, govStore, govBalance, show <$> viewStorage)
 
 
 callLigoEntrypoint ::
   ( CallCustomEp (VariantToParam var), HasBaseDAOEp (Parameter' (VariantToParam var))
   , MonadCleveland caps m
-  ) => ModelCall var -> TAddress (Parameter' (VariantToParam var)) () -> m ()
+  ) => ModelCall var -> ContractHandle (Parameter' (VariantToParam var)) (StorageSkeleton (VariantToExtra var)) () -> m ()
 callLigoEntrypoint mc dao = do
-  transferMoney (mc & mcSource & msoSender) oneMutez -- 0.9 XTZ
+  transfer (mc & mcSource & msoSender) oneMutez
   withSender (mc & mcSource & msoSender) $ case mc & mcParameter of
-    XtzAllowed (ConcreteEp (Propose p)) -> call dao (Call @"Propose") p
-    XtzAllowed (ConcreteEp (Transfer_contract_tokens p)) -> call dao (Call @"Transfer_contract_tokens") p
-    XtzAllowed (ConcreteEp (Transfer_ownership p)) -> call dao (Call @"Transfer_ownership") p
-    XtzAllowed (ConcreteEp (Accept_ownership p)) -> call dao (Call @"Accept_ownership") p
-    XtzAllowed (ConcreteEp (Default _)) -> call dao CallDefault ()
-
-    XtzForbidden (Vote p) -> call dao (Call @"Vote") p
-    XtzForbidden (Flush p) -> call dao (Call @"Flush") p
-    XtzForbidden (Freeze p) -> call dao (Call @"Freeze") p
-    XtzForbidden (Unfreeze p) -> call dao (Call @"Unfreeze") p
-    XtzForbidden (Update_delegate p) -> call dao (Call @"Update_delegate") p
-    XtzForbidden (Drop_proposal p) -> call dao (Call @"Drop_proposal") p
-    XtzForbidden (Unstake_vote p) -> call dao (Call @"Unstake_vote") p
+    XtzAllowed (ConcreteEp (Propose p)) -> transfer dao $ calling (ep @"Propose") p
+    XtzAllowed (ConcreteEp (Transfer_contract_tokens p)) -> transfer dao $ calling (ep @"Transfer_contract_tokens") p
+    XtzAllowed (ConcreteEp (Transfer_ownership p)) -> transfer dao $ calling (ep @"Transfer_ownership") p
+    XtzAllowed (ConcreteEp (Accept_ownership p)) -> transfer dao $ calling (ep @"Accept_ownership") p
+    XtzAllowed (ConcreteEp (Default _)) -> transfer dao
+
+    XtzForbidden (Vote p) -> transfer dao $ calling (ep @"Vote") p
+    XtzForbidden (Flush p) -> transfer dao $ calling (ep @"Flush") p
+    XtzForbidden (Freeze p) -> transfer dao $ calling (ep @"Freeze") p
+    XtzForbidden (Unfreeze p) -> transfer dao $ calling (ep @"Unfreeze") p
+    XtzForbidden (Update_delegate p) -> transfer dao $ calling (ep @"Update_delegate") p
+    XtzForbidden (Drop_proposal p) -> transfer dao $ calling (ep @"Drop_proposal") p
+    XtzForbidden (Unstake_vote p) -> transfer dao $ calling (ep @"Unstake_vote") p
```

Comment:
Gave this a try, but couldn't apply because of the [last branch](https://github.com/tezos-commons/baseDAO/blob/master/haskell/test/SMT/Common/Run.hs#L281) of the case statement.

Reply:
Ah, indeed, I missed that last case. Still possible, but the amount of fiddling with types in `CallCustomEp` is just not worth it at that point.